src/marked.js

```diff
@@ -164,6 +154,50 @@ marked.use = function(extension) {
   marked.setOptions(opts);
 };
 
+/**
+ * Iterate over every token
+ */
+
+marked.iterateTokens = function(tokens, callback) {
+  let ret;
+  for (const token of tokens) {
+    ret = callback(token);
+    if (ret === false) {
+      return false;
+    }
```

Comment:
if the callback function returns `false` when iterating over the tokens the iterating will stop.

Reply:
Is this a breaking change? AFAICT it would previously touch every token, and just eat failures?

Reply:
The async highlighting still iterates over every token since it doesn't ever return `false`. I just added this to the new `iterateTokens` function so there is a way to stop if the user wants to.

This behavior is similar to other library functions such as [jquery.each](https://api.jquery.com/each/).